### DISCUSSION TITLE
How to approach newer functions (detailer, control, ...) via the API?

### DISCUSSION THREAD

**Discussion by anothertal3** [vladmandic/sdnext#3513] [Q&A] [answered]
Hi everyone,

I've been using SD.Next via the API for some time now and managed to find the correct parameters settings for refining and extensions such as ADetailer or ControlNet.

After observing recent updates I've got the feeling it's time to move on and try to embrace the new built-in features such as "detailer" or "control" as they seem to be quite powerful and well-maintained.

However, I'm unsure about the best migration approach. 

I know that there's an online API documentation when using "--docs" which is a big help to get started. But it's often confusing to understand which API parameter matches which UI parameter or which data type is actually required. At least that was my experience when trying to get refining or extensions to work.

Are there per chance any examples or documentation about how to use those new features via the API? 

Thank you!

**Comment by vladmandic** [comment, marked as answer]
regarding control, there is api example in `cli/api-contro.py`.

detailer is simple, its just a checkbox in any generate request (txt2img, img2img, control) - `detailer: true`
detailer settings are in global server settings, so they can be set using `/sdapi/v1/options` api endpoint before calling generate.

**Reply by anothertal3** [reply]
Thank you
